Study landing pages

will-moore opened the pull request:
Work in Progress...

For study with Zarr data in BIA:

![Screenshot 2025-05-14 at 17 31 42](https://github.com/user-attachments/assets/cbb9c21b-04a3-44b4-8f50-f9d892ea484d)

For IDR study (download links to ftp site):

![Screenshot 2025-05-14 at 17 32 09](https://github.com/user-attachments/assets/32242b13-fa01-446c-aa65-fc83ee70d061)

will-moore commented:
NB: deployed on idr-testing, but not working yet. Maybe due to public permissions or something else...

Edited `sudo vi /opt/omero/web/config/00-omero-web.omero` to add `study/` to public URLs:

```
config set -- omero.web.public.url_filter '^/(api/|webadmin/myphoto/|mapr/|figure/|iviewer/|$|search/|search/cell/|search/tissue/|study/|gallery-api/|gallery_settings/|cell/|tissue/|webclient/(?!(action|annotate_(file|tags|comment|rating|map)|script_ui|ome_tiff|figure_script|render_image_download))|webgateway/(?!(archived_files|download_as)))'
```
But not working yet using `ssh -A idr-testing.openmicroscopy.org -L 1080:localhost:80` and `http://localhost:1080/study/idr0001/` (redirects)

will-moore commented:
The `Browse Image Data at the BioImage Archive` leads to https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/pages/S-BIAD866.html so you can browse there.
This isn't a zarr link that you'd want to copy elsewhere. That is a BIA site and we know that because the image clientpath starts with `https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/`, which is only true for the studies we updated with `mkngff` workflow.

will-moore commented:
@sbesson - The `/study/...` URLs are probably getting redirected due to: 
https://github.com/IDR/deployment/blob/f7e8bb9699a64cb291c8dde09f519bb7c0eaf756/ansible/group_vars/proxy-hosts.yml#L243

Any idea what this is supposed to handle?
Can we remove `study` from that list. If we want `/study/invalid` to redirect to 'home' we can handle that in idr-gallery if "invalid" doesn't match any study.

sbesson commented:
https://github.com/IDR/deployment/pull/237 direct links were defined for `idr0083`

pwalczysko commented:
Thanks for explaining @will-moore 

> This isn't a zarr link that you'd want to copy elsewhere. That is a BIA site and we know that because the image clientpath starts with https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/, which is only true for the studies we updated with mkngff workflow.

Are such studies on S3 ? I can see an S3 link in the Imported from section of IDR webclient  on an image from idr0011 ? If yes, do I get the link to S3 as a user via landing page ?

will-moore commented:
@pwalczysko Yes, they are on s3. As discussed, you can browse from https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/pages/S-BIAD866.html to a plate (e.g. https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/pages/S-BIAD866/013015ae-64f8-494c-85ee-b4e364960b75.html) and Copy S3 URL e.g. https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/S-BIAD866/013015ae-64f8-494c-85ee-b4e364960b75/013015ae-64f8-494c-85ee-b4e364960b75.zarr/0
but this doesn't work because of the trailing `/0` which shouldn't be added for a Plate. E-mailed BIA about it.

joshmoore commented:
Some high-level thoughts/next steps:
 * To make the Search gods happy, I'd suggest we add schema.org metadata into the header of these new pages, and likely, some links to *other* pages (gallery, webclient, etc.) to point to what we *want* searched. See https://signposting.org/FAIR/ for some background.
 * If it's possible to kick off a background activity, I could see adding dynamic statistics to the sidebar (like when first loading the gallery page) to catch people's attention.

pwalczysko commented:
On https://idr-testing.openmicroscopy.org/study/idr0012/, what is the "External ulr" block supposed to mean ?
Please note, the url there is not leading to anything reasonable (seems like a link to S3).

![Screenshot 2025-05-29 at 11 50 22](https://github.com/user-attachments/assets/2e764822-f6b1-46ff-8495-64c7c492337e)

pwalczysko commented:
On https://idr-testing.openmicroscopy.org/study/idr0043/ the number of annotation files is very large, and this is making the landing page left-hand column too long for comfort.


Suggestion:

1. either do not display all annotation files if there are too many or
2. .... not sure atm

pwalczysko commented:
@will-moore Thank you, I think this is a major improvement. I have some comments/questions/suggestions in
1. https://github.com/IDR/idr-gallery/pull/50#issuecomment-2919029636
2. https://github.com/IDR/idr-gallery/pull/50#issuecomment-2919035541
3. https://github.com/IDR/idr-gallery/pull/50#issuecomment-2919053725
4. https://github.com/IDR/idr-gallery/pull/50#issuecomment-2919056998
5. https://github.com/IDR/idr-gallery/pull/50#issuecomment-2919081785

pwalczysko reviewed (changes requested):
Please address 5 comments. Thank you, looks great in general

will-moore commented:
@pwalczysko I think I have addressed those 5 comments (and re-deployed on idr-testing):

Using the `Globus` link from BIA page (e.g. https://www.ebi.ac.uk/biostudies/BioImages/studies/S-BIAD1989) as an example, I actually created a direct link to the Globus web UI for the particular study, hopefully avoiding the need for Download instructions. But I also added a link to the Download page for FTP info etc.

@joshmoore I made a start on schema.org jsonld in the <head>, using terms from https://bioschemas.org/. E.g:

```
    <script type="application/ld+json">
        {
            "@context": "https://schema.org/",
            "@type": "Dataset",
            "name": "A genomic Multiprocess survey of machineries that control and link cell shape, microtubule organization, and cell-cycle progression.",
            "description": "\nPrimary screen of fission yeast knock out mutants looking for genes controlling cell shape, microtubules, and cell-cycle progression. 262 genes controlling specific aspects of those processes are identifed, validated, and functionally annotated.",
            "url": "https://idr.openmicroscopy.org/study/idr0001/",
            "license": "https://creativecommons.org/licenses/by/4.0/"
        }
    </script>
```
and to make it easier to access and review, you can also find the same content at e.g. https://idr-testing.openmicroscopy.org/study/idr0001/json/ (and there's a link to that from each page).

pwalczysko commented:
Typo on https://idr-testing.openmicroscopy.org/study/idr0079/

Globas -> Globus
![Screenshot 2025-05-30 at 15 40 34](https://github.com/user-attachments/assets/61067906-fddf-4d92-b808-8dee6f1ba877)

pwalczysko commented:
https://idr-testing.openmicroscopy.org/study/idr0099/
https://idr-testing.openmicroscopy.org/study/idr0106/

The Download section is still offering only the JSON (which is great). 
Would you please explain why ?
Happy if it is not possible.

pwalczysko reviewed (changes requested):
Please address https://github.com/IDR/idr-gallery/pull/50#issuecomment-2922598031.
Really like the way this is shaping in general though, thanks a lot for the other fixes.

pwalczysko commented:
Go to 
https://idr-testing.openmicroscopy.org/study/idr0011/
then click onto "Open in Bio-File-Finder"
https://idr-testing.openmicroscopy.org/biofilefinder/bff/app/?source=%7B%22uri%22%3A%20%22https%3A//idr-testing.openmicroscopy.org/biofilefinder/table/7091762/omero.parquet%22%2C%20%22type%22%3A%20%22parquet%22%2C%20%22name%22%3A%20%22omero_table.parquet%22%7D

Observe

![Screenshot 2025-06-09 at 10 07 50](https://github.com/user-attachments/assets/756b45c4-41a1-4231-9d3e-7b53c336fab0)

Is that the problem ^^^ you were mentioning during meeting @will-moore ?

will-moore commented:
Outstanding questions to resolve:

 - [ ] BFF is experimental and won't be included (I'll revert [7a165c7](https://github.com/IDR/idr-gallery/pull/50/commits/7a165c732a7fc02859cc0a551643fb110572b142) for now)
 - [ ] JSON (schema.org). Do we leave as is (kinda placeholder amount of data) or remove JSON link or add a bit more data?
 - [ ] Links from home page. Should these open in a new Tab? Will "shouldn't be the default behaviour as this forces that on the user, whereas with a regular link the user can choose to open in new tab. Also less friendly on mobile". Frances/Petr like new tabs - will test

pwalczysko commented:
On the idr0161 landing page https://idr-testing.openmicroscopy.org/study/idr0161/ there is the link

![Screenshot 2025-06-09 at 10 39 48](https://github.com/user-attachments/assets/31bf034f-3cbd-442c-8a7d-e6bd25f91185)


https://uk1s3.embassy.ebi.ac.uk/bia-integrator-data/pages/S-BIAD1257.html

will-moore commented:
@pwalczysko Thanks - yeah, that's a bug. It should link to https://www.ebi.ac.uk/biostudies/bioimages/studies/S-BIAD1257. I'll look into it...

pwalczysko commented:
> @pwalczysko Thanks - yeah, that's a bug. It should link to https://www.ebi.ac.uk/biostudies/bioimages/studies/S-BIAD1257. I'll look into it...

Confirming that this ^^^ (commant https://github.com/IDR/idr-gallery/pull/50#issuecomment-2955254409) is fixed

francesw commented:
@will-moore I'm ok with the opening of links in same browser tab or window when it's related to IDR as users can go back easily to IDR homepage.  However, when it comes to external links such as Publication DOI, License and External URL, it will be hard for users to navigate back to IDR homepage easily.  So perhaps if links are external to IDR, these should open in new browser or window?  Just a thought.

will-moore commented:
No download link for various studies that only link to companion files on github.
https://idr-testing.openmicroscopy.org/study/idr0021/
https://idr-testing.openmicroscopy.org/study/idr0099/
https://idr-testing.openmicroscopy.org/study/idr0106/

For these, we need to handle as for other "IDR" backed studies: link to Globus & FTP etc:
https://ftp.ebi.ac.uk/pub/databases/IDR/idr0021-lawo-pericentriolarmaterial/
https://ftp.ebi.ac.uk/pub/databases/IDR/idr0099-jain-beetlelightsheet/
https://ftp.ebi.ac.uk/pub/databases/IDR/idr0106-kubota-lunglightsheet/

joshmoore commented on idr_gallery/templates/idr_gallery/idr_study.html:
```diff
@@ -0,0 +1,182 @@
+                <p>Data for this study is available at the <a target="_blank" href="{{ bia_accession.link }}">BioImage Archive: {{ bia_accession.name }}</a>.</p>
+            {% endif %}
+
+            <p>Download as <a href="{% url 'idr_gallery_study_jsonld' idr_id %}" target="_blank">JSON</a>.</p>
```
In general, we probably need to make it clear what someone would use the json for. It's incomplete enough at the moment, that I don't think there's much an end user would want to do with it.

If we want to keep it for the moment, it might be in a more advanced section. "Below the fold" so to speak. Alternatively, we wait until it's complete enough to be shown.

joshmoore commented on idr_gallery/views.py:
```diff
@@ -105,6 +111,159 @@ def index(request, super_category=None, conn=None, **kwargs):
+    license = parse_kvp_with_link("License", kvps)
+    titles = kvps.get("Study Title", kvps.get("Publication Title"))
+    jsonld = {
+        "@context": "https://schema.org/",
```
These all seem good to have, so :+1: for keeping this in the page header. The next steps (after this PR) that we will likely want to look into are:
 - more fields (authors, etc.)
 - adding pointers from *other* pages to this page as the official source of machine parseable metadata
 - whether or not to go ahead and structure this json-ld into a RO-Crate and just offer that as the end point. (e.g., see https://data.emobon.embrc.eu/#crates)

francesw commented:
Confirming external links open in a new tab (and Data DOI doesn't as it is internal).

pwalczysko commented:
Happy with the new changes. Ready to merge fmpov.
Thank you very much.
